### PR TITLE
MM-23574: Remove Segment.

### PR DESCRIPTION
**Pull request by grundleborg**
#### Summary
Delete Segment.

@agnivade requesting your review because `go mod tidy` did more than I was expecting...

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-23574

### CODE REVIEW

File: app/server.go [hunk 1]
```diff
@@ -442,7 +440,7 @@ func (s *Server) Shutdown() error {
 		}
 	}
 
-	err := s.shutdownDiagnostics()
+	err := s.shutdownRudder()
```
Comment (reflog) [inline app/server.go:443]: i'm 1/4 on this, but I think we should lave the names as `shutdownDiagnostics` and `initDiagnostics`. Easier for people to read the code (not everyone knows what Rudder is)

**Comment by grundleborg** [comment]
@reflog agreed - I've updated it.

File: app/server.go [hunk 2]
```diff
@@ -125,9 +124,8 @@ type Server struct {
 	clientConfigHash    atomic.Value
 	limitedClientConfig atomic.Value
 
-	diagnosticId     string
-	diagnosticClient analytics.Client
-	rudderClient     rudder.Client
+	diagnosticId string
+	rudderClient rudder.Client
```
Comment (agnivade) [inline app/server.go:128]: Should we name it `diagnosticClient` to keep things same as previous ?

**Comment by lindy65** [comment]
Hi @grundleborg - is basic smoke testing ok for testing this PR or is there anything specific I should be testing for? (other than Segment being removed that is)

**Comment by grundleborg** [comment]
Quick smoke test should be fine. If it's reasonably straightforward, checking with Alex to make sure Server-side telemetry is still received via Rudder when you are testing this PR would also be great, but that can happen in master if needed.

**Comment by lindy65** [comment]
Thanks! I believe Rudder can only be checked after merge so we can check that out on `master` 👍

**Pull request merged**